x-pack/test/saved_object_api_integration/security_and_spaces/apis/find.ts

```diff
@@ -35,40 +36,107 @@ export default function ({ getService }: FtrProviderContext) {
   const esArchiver = getService('esArchiver');
 
   const { addTests, createTestDefinitions } = findTestSuiteFactory(esArchiver, supertest);
-  const createTests = (spaceId: string) => {
-    const { normalTypes, hiddenAndUnknownTypes, allTypes } = createTestCases(spaceId);
+  const createTests = (spaceId: string, user: TestUser) => {
+    const currentSpaceCases = createTestCases(spaceId, []);
+
+    const explicitCrossSpace = createTestCases(spaceId, ['default', 'space_1', 'space_2']);
+    const wildcardCrossSpace = createTestCases(spaceId, ['*']);
+
+    if (user.username === 'elastic') {
+      return {
+        currentSpace: createTestDefinitions(currentSpaceCases.allTypes, false, { user }),
+        crossSpace: createTestDefinitions(explicitCrossSpace.allTypes, false, { user }),
+      };
+    }
+
+    const authorizedAtCurrentSpace =
+      user.authorizedAtSpaces.includes(spaceId) || user.authorizedAtSpaces.includes('*');
+
+    const authorizedExplicitCrossSpaces = ['default', 'space_1', 'space_2'].filter(
+      (s) =>
+        user.authorizedAtSpaces.includes('*') ||
+        (s !== spaceId && user.authorizedAtSpaces.includes(s))
+    );
+
+    const authorizedWildcardCrossSpaces = ['default', 'space_1', 'space_2'].filter(
+      (s) => user.authorizedAtSpaces.includes('*') || user.authorizedAtSpaces.includes(s)
+    );
+
+    const explicitCrossSpaceDefinitions =
+      authorizedExplicitCrossSpaces.length > 0
+        ? [
+            createTestDefinitions(explicitCrossSpace.normalTypes, false, { user }),
+            createTestDefinitions(
+              explicitCrossSpace.hiddenAndUnknownTypes,
```

Comment:
I swear I tried, multiple times, but I'm still drowning in your security/spaces integration tests 😅

Reply:
Same 😹

Reply:
Same 😬